Migration detail

priley86 opened the pull request:
Starts the migration plan detail view. Still a W.I.P. (needs pagination and filters). Posting this for initial review. Getting so close... 😸 

- [x] Adds actions/reducers for retrieving plan migration detail (plan request/tasks)
- [x] Moves Breadcrumbs into views so they can customized easily
- [x] Adds sub url route for plan migration detail `/migration/plan/:id`
- [x] Wires migration in progress card link to plan migration detail page passing id
- [x] Adds list view for displaying migration detail
- [x] Wires hostname, started/elapsed times, status, and disk completion utilization bar
- [x] Disk utilization bar is calculated across all disks for the VM task.
- [x] started/elapsed times are calculated similarly to the migration in progress card (moved logic into a helper function). Tested the start/elapsed times are ISO/internationalized correctly for GMT.
- [x] Handles Migrating, Migration complete status.

- [x] Needs to handle other migration statuses listed in api notes (Pending, Validating, Pre-migration).
- [x] Add pagination
- [x] Add toolbar/filters
- [x] Add error/loading state messaging
- [x] rebase @michaelkro 's latest overview changes

![migration-details-min](https://user-images.githubusercontent.com/4237045/38479278-01c3e7d8-3b8c-11e8-83ed-3e1f1302930d.gif)

michaelkro commented on app/javascript/react/screens/App/Overview/components/Cards/MigrationsInProgressCard/MigrationInProgressCard.js:
```diff
@@ -42,34 +42,18 @@ const MigrationInProgressCard = ({ migration }) => {
-      </strong>{' '}
-      {__('Migrated')}
-    </span>
+    <a href={`/migration/plan/${migration.id}`}>
```
Can use `<Link>` to avoid full page refresh when transitioning

michaelkro commented:
Future nice to have:  Integrate `react-router-dom` with breadcrumbs to avoid full page refreshes

@vconzola @serenamarie125, other than the breadcrumbs, we do not have any navigation built into the migration detail page.  Do we want to add a `back` button?

AllenBW commented on app/javascript/react/screens/App/Plan/Plan.js:
```diff
@@ -0,0 +1,95 @@
+          <Spinner loading={isFetchingPlanRequests}>
+            {!isFetchingPlanRequests &&
+              !isRejectedPlanRequests &&
+              planRequestDetails.tasks &&
```
Consider adding tasks as an empty array to the `defaultProps` object.  It provides more default props context and simplifies the logic here, no need to check if it is defined, only that the length is > 0

michaelkro commented on app/javascript/react/screens/App/Plan/PlanReducer.js:
```diff
@@ -0,0 +1,86 @@
+        .set('planRequestDetails', {})
+        .set('isRejectedPlanRequests', false)
+        .set('isFetchingPlanRequests', false);
+    }
```
Also reset `errorPlanRequests` when request is fulfilled

vconzola commented:
@michaelkro Re: Back button on the plan details page, I don't think I've seen that in any of our other products, but I defer to @serenamarie125. She's much more familiar with than I am.

AllenBW commented on app/javascript/react/screens/App/Plan/components/PlanRequestDetailList.js:
```diff
@@ -0,0 +1,81 @@
+          additionalInfo={[
+            <div style={{ paddingRight: 60 }}>
+              <span>
+                <strong>Started: </strong>
```
Translation support (here and well and other places)

priley86 replied:
yup yup...added

priley86 commented:
thanks guys 👍 

i'll try to address these later today along with the other feature updates...

priley86 commented:
tried to address these comments first (see latest commits)...still working on the Toolbar/Paging integration today but hoping to wrap it up.

priley86 commented:
* rebased
* implemented sort and filters. I had to add a "natural sort" algorithm to ensure that `rhvh02.example.com` would sort before `rhvh11.example.com`. Will show show this in a gif shortly.

Still working on paging...

priley86 commented:
* finished paging
* adds error/warning empty states
* move popover into field level help
* slightly reduce margins/paddings in list view

should be good to go for the initial detail view...would like to iterate in future PRs so others can rebase.

cc: @michaelkro @serenamarie125 @vconzola 

Screenshots...

Filters 
![filter-detail](https://user-images.githubusercontent.com/4237045/38655850-e220393a-3de4-11e8-9473-3c987f7290b7.gif)

Sorting
![sorting-detail](https://user-images.githubusercontent.com/4237045/38655857-ed77f0e8-3de4-11e8-88fe-c64f7d49027d.gif)

Pagination
![paging-detail](https://user-images.githubusercontent.com/4237045/38655865-f2d7a038-3de4-11e8-9cd9-f094c8b7b1a9.gif)

Popovers
![popover-detail](https://user-images.githubusercontent.com/4237045/38655896-048c0d32-3de5-11e8-905e-7ed2ecaa8108.gif)

Error state
<img width="1391" alt="screen shot 2018-04-11 at 11 24 10 pm" src="https://user-images.githubusercontent.com/4237045/38655913-1d03e9e8-3de5-11e8-947a-12cd442be080.png">

Warning State
<img width="1197" alt="screen shot 2018-04-11 at 11 17 57 pm" src="https://user-images.githubusercontent.com/4237045/38655921-24b2009e-3de5-11e8-881c-9bad63fd56f4.png">

priley86 commented:
pushed a slight tweak to prevent updates from firing after polling... still seeing a few weird things though... testing..

priley86 commented:
one other thing that is missing here and @michaelkro raised. The user can currently filter two statuses at the same time, but that's just b/c i copied the default filter logic in PatternFly React. I think this can be addressed next sprint as a nice to have...

![screen shot 2018-04-12 at 12 24 26 pm](https://user-images.githubusercontent.com/4237045/38690664-a748fc94-3e4c-11e8-9e5b-585b60714acc.png)
...

bthurber commented:
https://bugzilla.redhat.com/show_bug.cgi?id=1564242
